### PR TITLE
fix: refactor linea goerli testnet implementation

### LINKED ISSUE #19331: 6
https://github.com/Uniswap/token-list-bridge-utils/pull/3

### PR DESCRIPTION
**Pull request by VGau**
## Explanation

- Refactored Linea Goerli testnet implementation
- Added new Linea Goerli infura rpc url : `https://linea-goerli.infura.io/v3/`
- Created new `LINEA_GOERLI` network type
- The Linea Goerli testnet is using infura rpc provider now instead of custom rpc provider

### Before
Screenshot 2023-06-05 at 1 23 50 PM

### After
Screenshot 2023-06-05 at 1 16 41 PM

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: shared/constants/network.ts [hunk 1]
```diff
@@ -267,10 +271,9 @@ export const TEST_NETWORK_TICKER_MAP: {
   [NETWORK_TYPES.SEPOLIA]: `${typedCapitalize(NETWORK_TYPES.SEPOLIA)}${
     CURRENCY_SYMBOLS.ETH
   }`,
-  [NETWORK_TYPES.LINEA_TESTNET]:
-    `Linea${CURRENCY_SYMBOLS.ETH}` as `${Capitalize<
-      typeof NETWORK_TYPES.LINEA_TESTNET
-    >}${typeof CURRENCY_SYMBOLS.ETH}`,
+  [NETWORK_TYPES.LINEA_GOERLI]: `Linea${CURRENCY_SYMBOLS.ETH}` as `${Capitalize<
```
Comment (Gudahtt) [inline shared/constants/network.ts:274]: Nit: rather than use a type cast here, we could update the return type to just be `string`. I don't see any purpose in having the type misrepresent the actual value like this - it's misleading and unnecessary.

File: shared/constants/network.ts [hunk 2]
```diff
@@ -416,10 +420,10 @@ export const ETHERSCAN_SUPPORTED_NETWORKS = {
     }`,
     networkId: CHAIN_ID_TO_NETWORK_ID_MAP[CHAIN_IDS.SEPOLIA],
   },
-  [CHAIN_IDS.LINEA_TESTNET]: {
-    domain: 'linea.build',
-    subdomain: 'explorer.goerli',
-    networkId: CHAIN_ID_TO_NETWORK_ID_MAP[CHAIN_IDS.LINEA_TESTNET],
+  [CHAIN_IDS.LINEA_GOERLI]: {
+    domain: 'lineascan.build',
```
Comment (Gudahtt) [inline shared/constants/network.ts:424]: I understand that there is a plan to switch to this block tracker, but I don't think we should make this change until the site is live

File: ui/pages/settings/networks-tab/networks-tab.constants.js [hunk 3]
```diff
@@ -45,13 +44,16 @@ const defaultNetworksData = [
...
-    ticker: TEST_NETWORK_TICKER_MAP[NETWORK_TYPES.LINEA_TESTNET],
-    blockExplorerUrl: 'https://explorer.goerli.linea.build',
+    providerType: NETWORK_TYPES.LINEA_GOERLI,
+    rpcUrl: getRpcUrl({
+      network: NETWORK_TYPES.LINEA_GOERLI,
+      excludeProjectId: true,
+    }),
+    chainId: CHAIN_IDS.LINEA_GOERLI,
+    ticker: TEST_NETWORK_TICKER_MAP[NETWORK_TYPES.LINEA_GOERLI],
+    blockExplorerUrl: 'https://goerli.lineascan.build',
```
Comment (Gudahtt) [inline ui/pages/settings/networks-tab/networks-tab.constants.js:56]: Likewise for this URL - we can wait until it's live

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js [hunk 4]
```diff
@@ -99,7 +101,7 @@ const NetworksListItem = ({
           />
         )
       )}
-      {network.isATestNetwork && (
+      {network.isATestNetwork && network.chainId !== CHAIN_IDS.LINEA_GOERLI && (
```
Comment (Gudahtt) [inline ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:104]: What is the purpose of this condition?

File: development/states/navigate-txs.json [hunk 5]
```diff
@@ -305,7 +305,7 @@
       "mainnet": "ok",
       "goerli": "ok",
       "sepolia": "ok",
-      "lineatestnet": "ok"
+      "lineaGoerli": "ok"
```
[the remark this answers is not in the document]
Comment (Gudahtt) [reply development/states/navigate-txs.json:308]: I have removed this file in a separate PR: #19597

File: ui/pages/settings/networks-tab/networks-tab.constants.js (hunk 3 above)
Comment (VGau) [reply ui/pages/settings/networks-tab/networks-tab.constants.js:56]: This block explorer is working :)

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js (hunk 4 above)
Comment (Gudahtt) [reply ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:104]: Got it, thanks.

It would be great to refactor this to make it more clear that we're trying to render a single icon per network. The way it's setup now is a bit hard to follow. That can wait for another time though

**Review by Gudahtt — DISMISSED** [review]
LGTM!

File: shared/constants/network.ts [hunk 6]
```diff
@@ -243,34 +246,31 @@ export const INFURA_PROVIDER_TYPES = [
...
   capitalize(k) as Capitalize<typeof k>;
 
 export const TEST_NETWORK_TICKER_MAP: {
-  [K in Exclude<
-    NetworkType,
-    'localhost' | 'mainnet' | 'rpc'
-  >]: `${Capitalize<K>}${typeof CURRENCY_SYMBOLS.ETH}`;
+  [K in Exclude<NetworkType, 'localhost' | 'mainnet' | 'rpc'>]:
+    | `${Capitalize<K>}${typeof CURRENCY_SYMBOLS.ETH}`
+    | string;
```
Comment (Gudahtt) [inline shared/constants/network.ts:265]: Nit: this type union can be simplified (the template is a subset of `string`, so this effectively collapses to `string`)
```suggestion
  [K in Exclude<NetworkType, 'localhost' | 'mainnet' | 'rpc'>]: string;
```

**Review by Gudahtt — DISMISSED** [review]

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js (hunk 4 above)
Comment (VGau) [reply ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:104]: Yes I agree :) Thanks

**Review by Gudahtt — APPROVED** [review]

**Review by danjm — APPROVED** [review]
LGTM
